spinalcordtoolbox/reports/qc.py

```diff
@@ -330,9 +330,9 @@ def _make_QC_image_for_3d_volumes(self, img, mask, slice_orientation):
         # if axial mosaic restrict width
         if slice_orientation == 'Axial':
             size_fig = [5, 5 * img.shape[0] / img.shape[1]]  # with dpi=300, will give 1500pix width
-        # if sagittal orientation restrict height
+        # if sagittal, set fixed size (to be stable for images with sagittal as well as axial orientation)
         elif slice_orientation == 'Sagittal':
-            size_fig = [5 * img.shape[1] / img.shape[0], 5]
+            size_fig = [5, 5]
```

Comment:
During [your investigation](https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3055#issuecomment-1256114262), you mentioned:

> This is fine if the input image has a sagitall orientation (and thus img.shape[0] is same (or similar) to img.shape[1], e.g. 896x896):

If it's true that we can safely assume that the image dimensions are similar, then:

* Scaling by `img.shape` is practically unnecessary, since `img.shape[1] / img.shape[0] ~= 1`.
* If that's true, then the `'Axial'` case could possibly also be reduced to `[5, 5]`? 
* If )that's_ true, then the `slice_orientation` parameter is entirely unnecessary for this method, since `size_fig = [5,5]` in both cases. 

This has pretty big implications, and it makes me wonder why we even make the distinction between `'Sagittal'` and `'Axial'` in the first place.

Reply:
The above makes me want to step back a moment and re-evaluate the root problem here:

1. Every QC report function is assumed to have one (and only one) valid orientation (via the `plane` variable).
2. `sct_label_vertebrae` expects `'Sagittal'` images, yet an `'Axial'` image was passed. ((In other words, the assumption from 1. was broken.))

If it's true that we've been making a faulty assumption here, then it makes me want to investigate whether there is some other way we can organize our QC code to avoid making this assumption in the first place.

I don't have a concrete suggestion yet -- I want to first try to better understand how our QC code is organized. I just wanted to quickly suggest that the problem/solution here might be a little more complicated than previously thought, and so I would want to hold off on merging for now?

Reply:
> * Scaling by `img.shape` is practically unnecessary, since `img.shape[1] / img.shape[0] ~= 1`.

Yes, it looks that for `'Sagittal'` QC (used for example for `sct_label_vertebrae` or `sct_label_utils`), the scaling by `img.shape` is unnecessary. The scaling had a minimal effect on the sagittally acquired images. But the scaling caused bad QC .png image size for axially acquired images. That's why I removed the scaling and set the `size_fig` to fix values [5, 5] in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3889/commits/dc77406270ed07f35771d96316fd06426e509ad0

> * If that's true, then the `'Axial'` case could possibly also be reduced to `[5, 5]`?

However, for `'Axial'` QC (used for example `sct_deepseg_sc` or `sct_deepseg_gm`), it looks that the scaling is necessary. Otherwise, QC .png image has a high height and is visualised wrongly. See below.
Without scaling:
<img width="500" alt="Screenshot 2022-09-24 at 19 18 52" src="https://user-images.githubusercontent.com/39456460/192112430-0679ab86-bcda-4c68-91b0-84abda19dda7.png">

With scaling:
<img width="500" alt="Screenshot 2022-09-24 at 19 19 00" src="https://user-images.githubusercontent.com/39456460/192112438-c819ab27-21a4-4a2c-a11d-5309e077303d.png">

Reply:
> The above makes me want to step back a moment and re-evaluate the root problem here:
> 
> 1. Every QC report function is assumed to have one (and only one) valid orientation (via the `plane` variable).

I think that this is fine. For example, for `sct_label_vertebrae` or `sct_labels_utils`, it makes sense to create only the `'Sagittal'` QC (because from the `'Axial'` it would not be possible to assess the labeling). 
Similarly, for `sct_deepseg_sc` or `sct_deepseg_gm`, it makes sense to create only the `'Axial'` QC (again, the reason is that it is impractical to assess the SC or GM segmentation in other than the axial plane).

> 2. `sct_label_vertebrae` expects `'Sagittal'` images, yet an `'Axial'` image was passed. ((In other words, the assumption from 1. was broken.))

I think that the terms `'Sagittal'` and `'Axial'` in the context of the QC do not necessarily imply the expected orientation of the input images. Rather, it defines how the QC .png images will be created. This makes sense for me, for example, regardless if the input image has sagittal or axial orientation, the QC for `sct_label_vertebrae` should be `'Sagittal'` to allow us to assess the labeling.

Reply:
Ah! Thank you for clarifying. Your changes make perfect sense to me now. :)

I'd like to add a few extra comments just to clarify exactly the details you've mentioned. But for the most part, this LGTM. :)